internal/resource.go

```diff
@@ -50,90 +54,127 @@ func getUrl(u string, fileName string, ctx context.Context) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("invalid url '%s': %s", u, err)
 	}
-	log.Debug().Str("URL", u).Msg("Downloading")
-	err = requests.
-		URL(u).
+
+	req := requests.URL(u).
 		Header("Accept", "*/*").
-		ToFile(fileName).
-		Fetch(ctx)
+		ToFile(fileName)
+
+	// Add authorization header if the URL matches the CacheUri and ARTIFACTORY_TOKEN is set.
+	token := os.Getenv("ARTIFACTORY_TOKEN")
+	if token != "" {
+		authHeader := "Bearer " + token
+		req = req.Header("Authorization", authHeader)
```

Comment:
I don't think you want to add this to *every* Grabit download -- you only want to add it to cache downloads.

Otherwise, you're sending your Artifactory credential string to the web server where you're downloading from!